Sources/SwiftUIKit/Views/Map.swift

```diff
@@ -0,0 +1,453 @@
+//
+//  Map.swift
+//  SwiftUIKit
+//
+//  Created by Oskar on 12/04/2020.
+//
+
+import Foundation
+import MapKit
+
+public class Map: MKMapView {
+  
+  fileprivate var initialCoordinates: CLLocationCoordinate2D
+  
+  fileprivate lazy var onFinishLoadingHandler: ((MKMapView) -> ())? = nil
+  
+  fileprivate lazy var afterRegionChangeHandler: ((MKMapView) -> ())? = nil
+  
+  fileprivate lazy var beforeRegionChangeHandler: ((MKMapView) -> ())? = nil
+  
+  fileprivate lazy var annotationViewConfigurationHandler: ((MKAnnotationView?, MKAnnotation) -> (MKAnnotationView?))? = nil
+  
+  fileprivate lazy var onAccessoryTapHandler: ((MKMapView, MKAnnotationView, UIControl) -> ())? = nil
+  
+  fileprivate lazy var onAnnotationViewStateChangeHandler: ((MKMapView, MKAnnotationView, MKAnnotationView.DragState, MKAnnotationView.DragState) -> ())? = nil
+  
+  fileprivate lazy var onAnnotationSelectHandler: ((MKMapView, MKAnnotationView) -> ())? = nil
+  
+  fileprivate lazy var onAnnotationDeselectHandler: ((MKMapView, MKAnnotationView) -> ())? = nil
+  
+  fileprivate lazy var annotationViewIdentifier: String? = nil
+  
+  public init(lat latitude: Double,
+              lon longitude: Double,
+              annotations: (() -> [(latitude: CLLocationDegrees,
+                                    longitude: CLLocationDegrees,
+                                    title: String?,
+                                    subtitle: String?)])? = nil) {
+    
+    let coordinates = CLLocationCoordinate2D(latitude: latitude, longitude: longitude)
+    
+    initialCoordinates = coordinates
+    
+    super.init(frame: .zero)
+    
+    let span = MKCoordinateSpan(latitudeDelta: region.span.latitudeDelta / 2,
+                                longitudeDelta: region.span.longitudeDelta / 2)
+    
+    let region = MKCoordinateRegion(center: coordinates, span: span)
+    setRegion(region, animated: true)
+    
+    if let annotations = annotations {
+      DispatchQueue.global().async { [weak self] in
+        guard let self = self else { return }
+        
+        for (latitude, longitude, title, subtitle) in annotations() {
+          let coordinate = CLLocationCoordinate2D(latitude: latitude,
+                                                  longitude: longitude)
+          let point = MKPointAnnotation()
+          point.coordinate = coordinate
+          point.title = title
+          point.subtitle = subtitle
+          
+          DispatchQueue.main.async {
+            self.addAnnotation(point)
+          }
+        }
+      }
+    }
```

Comment:
Could we have an `addAnnotations` function? 

```
func add(annotation: MKAnnotation) -> Self
func add(point: MapPoint) -> Self
func add(annotations: [MKAnnotation]) -> Self
func add(points: [MapPoint]) -> Self
```

Reply:
[Done](https://github.com/0xLeif/SwiftUIKit/pull/132/commits/033235dc6f8b8df80383366aede5dcd892cff4d7)